lxd/container_lxc.go

```diff
@@ -3406,9 +3406,20 @@ func (c *containerLXC) Migrate(cmd uint, stateDir string, function string, stop
 		 * namespace.
 		 */
 		if !c.IsPrivileged() {
-			if err := c.IdmapSet().ShiftRootfs(stateDir); err != nil {
+			err = c.StorageStart()
+			if err != nil {
 				return err
 			}
+
+			err = c.IdmapSet().ShiftRootfs(stateDir)
+			err2 := c.StorageStop()
+			if err != nil {
+				return err
+			}
+
+			if err2 != nil {
+				return err2
+			}
```

Comment:
I don't think it really matters what order their in; it's just a question of which error is more useful to surface. I thought it would be the shift error, but I don't really mind either way and I can reorder them if you want.